src/pysdmx/api/gds/__init__.py

```diff
@@ -0,0 +1,338 @@
+
+    def _sdmx_api_q(self, id: str = REST_ALL) -> GdsQuery:
+        return GdsQuery(artefact_type=GdsType.GDS_SDMX_API, resource_id=id)
+
+    def _services_q(
+        self, agency: str, resource: str = REST_ALL, version: str = REST_ALL
+    ) -> GdsQuery:
+        return GdsQuery(
+            artefact_type=GdsType.GDS_SERVICE,
+            agency=agency,
+            resource_id=resource,
+            version=version,
+        )
+
+    def _urn_resolver_q(self, urn: str) -> GdsQuery:
+        return GdsQuery(
+            artefact_type=GdsType.GDS_URN_RESOLVER, resource_id=urn
+        )
+
+
+class GdsClient(__BaseGdsClient):
+    """A client to be used to retrieve metadata from the GDS.
+
+    With this client, metadata will be retrieved in a synchronous fashion.
+    """
+
+    def __init__(
+        self,
+        api_endpoint: str = GDS_BASE_ENDPOINT,
+        pem: Optional[str] = None,
+    ) -> None:
+        """Instantiate a new client against the target endpoint.
+
+        Args:
+            api_endpoint: The endpoint of the targeted service.
+            pem: In case the service exposed a certificate created
+                by an unknown certificate authority, you can pass
+                a pem file for this authority using this parameter.
+        """
+        super().__init__(api_endpoint, pem)
+        self.__service = GdsRestService(
+            self.api_endpoint,
+            pem=pem,
+            timeout=10.0,
+        )
+
+    def __fetch(
+        self,
+        query: GdsQuery,
+    ) -> bytes:
+        """Fetch the requested metadata from the GDS service."""
+        return self.__service.gds(query)
+
+    def get_agencies(self, agency: str) -> Sequence[Agency]:
+        """Get the list of agencies for the supplied name.
+
+        Args:
+            agency: The agency maintaining the agency scheme from
+                which sub-agencies must be returned.
+
+        Returns:
+            The requested list of agencies.
+        """
+        query = super()._agencies_q(agency)
+        out = self.__fetch(query)
+        agencies = super()._out(out, self.reader.agencies)
+        return agencies
+
+    def get_catalogs(
+        self,
+        catalog: str,
+        resource: str = REST_ALL,
+        version: str = REST_ALL,
+        resource_type: Optional[Literal["data", "metadata"]] = None,
+        message_format: Optional[Literal["json", "csv", "xml"]] = None,
+        api_version: Optional[str] = None,
+        detail: Optional[Literal["full", "raw"]] = None,
+        references: Optional[Literal["none", "children"]] = None,
+    ) -> Sequence[GdsCatalog]:
+        """Get the list of catalogs for the supplied parameters.
+
+        Args:
+            catalog: The agency maintaining the catalog.
+            resource: The resource ID(s) to query. Defaults to '*'.
+            version: The version(s) of the resource. Defaults to '*'.
+            resource_type: The type of resource (e.g., 'data', 'metadata').
```

Comment:
A couple of suggestions on the wording:
`resource_type: Filters the endpoints that support the requested resource type (eg, 'data', 'metadata')`